lib/segment/src/segment.rs

```diff
@@ -1112,10 +1112,15 @@ impl SegmentEntry for Segment {
                 (key, PayloadIndexInfo::new(index_schema, points_count))
             })
             .collect();
+        let num_vectors = self
+            .vector_data
+            .values()
+            .map(|data| data.vector_storage.borrow().available_vector_count())
+            .sum();
 
         SegmentInfo {
             segment_type: self.segment_type,
-            num_vectors: self.available_point_count() * self.vector_data.len(),
+            num_vectors,
```

Comment:
Is it possible to add a simple unit test showing that the count is now more precise?

Reply:
Good idea. Will add, maybe as integration test instead.